stopes/pipelines/asr_bleu/retrieve_data.py

```diff
@@ -0,0 +1,138 @@
+import os
+import logging
+import typing as tp
+import pandas as pd
+from dataclasses import dataclass
+from pathlib import Path
+from glob import glob
+from omegaconf.omegaconf import MISSING
+
+from stopes.core.launcher import Launcher
+from stopes.core.stopes_module import Requirements, StopesModule
+
+@dataclass
+class RetrieveDataJob:
+    audio_path: str = MISSING
+    reference_path: str = MISSING
+    audio_format: str = MISSING
+    reference_format: str = MISSING
+    reference_tsv_column: tp.Optional[str] = None
+
+@dataclass
+class RetrieveDataConfig:
+    retrieve_data_jobs: tp.List[RetrieveDataJob] = MISSING
+
+class RetrieveData(StopesModule):
+    def __init__(self, 
+                 config: RetrieveDataConfig
+    ):
+        print(config)
+        super().__init__(config=config, config_class=RetrieveDataConfig)
+
+    def array(self):
+        return self.config.retrieve_data_jobs
+
+    def requirements(self) -> Requirements:
+        return Requirements(
+            nodes=1,
+            tasks_per_node=1,
+            gpus_per_node=0,
+            cpus_per_task=1,
+            timeout_min=24 * 60,
+        )
+    
+    def _extract_audio_for_eval(self, audio_dirpath: str, audio_format: str):
+        """Extract audio file paths for subsequent transcription"""
+        if audio_format == "n_pred.wav":
+            """
+            The assumption here is that 0_pred.wav corresponds to the reference at line position 0 from the reference manifest
+            """
+            audio_list = []
+            audio_fp_list = glob((Path(audio_dirpath) / "*_pred.wav").as_posix())
+            audio_fp_list = sorted(
+                audio_fp_list, key=lambda x: int(os.path.basename(x).split("_")[0])
```

Comment:
We will assume that audios and reference_text in the input configuration are pairs. Will sorting by basename here disrupt this matching?

Reply:
Could you explain more about your concern? My understanding is that basename is just the isolated file name (without the rest of the path). The audio files are named `n_pred.wav`, then they're sorted by their number `n`. As long as there are the same number of reference lines as audio files, and the names are formatted properly, how would the matching get disrupted?

Reply:
then, nvm, it looks like the matching will be correct as long as the input reference rows with the same order as the wav name. Actually, the generation part (before the ASR-BLEU) would generate the *_pred.wav according to the reference column and the order would be helpful for debugging.